internal/testutils/stats/test_metrics_recorder.go

```diff
@@ -75,6 +75,15 @@ func (r *TestMetricsRecorder) AssertDataForMetric(metricName string, wantVal flo
 	}
 }
 
+// AssertNoDataForMetric asserts no data is present for metric.
+func (r *TestMetricsRecorder) AssertNoDataForMetric(metricName string) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	if _, ok := r.data[estats.Metric(metricName)]; ok {
+		r.t.Fatalf("Data is present for metric %v", metricName)
```

Comment:
This is following an anti-pattern: https://google.github.io/styleguide/go/decisions#assert

Now that we have more time, let's address this.

Reply:
Didn't pass t to this component anymore, did all the assertions inline.